digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/JunctionDAO.scala

```diff
@@ -167,21 +169,27 @@ class JunctionDAO extends BaseDAO {
          WHERE j.VALID_TO IS NULL AND j.END_DATE IS NULL AND j.NODE_NUMBER IS NULL
            AND rw.ROADWAY_NUMBER IN (${roadwayNumbers.mkString(", ")})
        """
-    queryListTemplate(query)
+      queryListTemplate(query)
+    } else {
+      Seq.empty[JunctionTemplate]
+    }
   }
 
-  def fetchTerminatedByRoadwayNumbers(roadwayNumbers: Iterable[Long]) : Seq[Junction] = {
-    val query =
-      s"""
+  def fetchHistoryByRoadwayNumbers(roadwayNumbers: Iterable[Long]) : Seq[Junction] = {
+    if (roadwayNumbers.nonEmpty) {
+      val query =
+        s"""
          SELECT j.ID, j.JUNCTION_NUMBER, j.NODE_NUMBER, j.START_DATE, j.END_DATE, j.VALID_FROM, j.VALID_TO, j.CREATED_BY, j.CREATED_TIME
          FROM JUNCTION j
-         LEFT JOIN JUNCTION_POINT jp ON j.ID = jp.JUNCTION_ID AND jp.VALID_TO IS NULL
```

Comment:
Why rows with valid_to set should be returned? If valid_to is set, the row is logically deleted.

Reply:
Okay, this is only used from the tests. Renaming this to fetchExpiredByRoadwayNumbers.